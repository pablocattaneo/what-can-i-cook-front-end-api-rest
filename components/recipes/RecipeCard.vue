<template>
  <b-card
    :border-variant="borderVariant"
    img-alt="Image"
    img-top
    tag="article"
    no-body
    class="w-100"
  >
    <b-card-header v-if="showCardHeader" header-tag="header">
      <ul class="nav nav-pills card-header-pills">
        <li class="nav-item">
          <nuxt-link
            :to="
              localePath({
                path: `/admin/recipes/${id}`
              })
            "
            class="btn btn-link"
            >{{ $t('actions.edit') }}</nuxt-link
          >
        </li>
        <li class="nav-item">
          <b-button @click="deleteRecipe(id)" variant="link">{{
            $t('actions.delete')
          }}</b-button>
        </li>
      </ul>
    </b-card-header>
    <b-img-lazy
      v-if="bImgLazySrc"
      :src="bImgLazySrc"
      :alt="title"
      :title="title"
      :blank-src="null"
    />
    <b-card-body>
      <b-card-title>{{ title }}</b-card-title>
      <b-card-text v-if="description">
        {{ description }}
      </b-card-text>
    </b-card-body>
    <b-list-group v-if="showMoreInfo" flush>
      <template v-for="(info, key, index) in moreInfo">
        <b-list-group-item v-if="info" :key="index + key"
          >{{ $t(`recipes.${key}`) }}: {{ info }}</b-list-group-item
        >
      </template>
    </b-list-group>
    <template v-slot:footer>
      <n-link to="/" variant="primary">{{ $t('Read_more') }}</n-link>
    </template>
  </b-card>
</template>

<script>
import wcAuthenticationMixin from '@/mixins/wc-authentication-mixin.js'

export default {
  mixins: [wcAuthenticationMixin],
  props: {
    id: {
      type: [String, Number],
      default: null
    },
    title: {
      type: String,
      default: ''
    },
    bImgLazySrc: {
      type: String,
      default: ''
    },
    description: {
      type: String,
      default: ''
    },
    moreInfo: {
      type: Object,
      default: () => ({})
    },
    directions: {
      type: Array,
      default: () => []
    },
    borderVariant: {
      type: String,
      default: 'primary'
    },
    showCardHeader: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    showMoreInfo() {
      return Object.keys(this.moreInfo).some(
        (key) => this.moreInfo[key] !== null
      )
    }
  },
  methods: {
    async deleteRecipe(recipeId) {
      await this.authenticate()
      await this.$axios.$delete(`/recipe/${recipeId}`)
      this.$emit('recipeDeleted')
    }
  }
}
</script>

<style lang="stylus" scoped>
img
  width 100%
</style>
