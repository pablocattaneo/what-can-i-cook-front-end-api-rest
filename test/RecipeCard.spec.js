import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils'

import RecipeCard from '@/components/recipes/RecipeCard'
import { BootstrapVue } from 'bootstrap-vue'

const localVue = createLocalVue()

let wrapper
beforeEach(() => {
  localVue.use(BootstrapVue)
  wrapper = shallowMount(RecipeCard, {
    localVue,
    mocks: {
      $t: () => {},
      localePath: () => ''
    },
    stubs: {
      'n-link': RouterLinkStub
    },
    propsData: {
      readMoreLink: '/',
      bImgLazySrc: '/'
    }
  })
})

describe('Snaphots', () => {
  test('Branch: Default. It verify if non-intentional changes were made.', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('deleteRecipe method', () => {
  test('deleteRecipe method should be definded', () => {
    expect(wrapper.vm.deleteRecipe).toBeDefined()
  })
})

describe('Props', () => {
  test('The component is well formed so prop id should exist', () => {
    expect(wrapper.props()).toHaveProperty('id')
  })

  test('The component is well formed so prop title should exist', () => {
    expect(wrapper.props()).toHaveProperty('title')
  })

  test('The component is well formed so prop bImgLazySrc should exist', () => {
    expect(wrapper.props()).toHaveProperty('bImgLazySrc')
  })

  test('The component is well formed so prop description should exist', () => {
    expect(wrapper.props()).toHaveProperty('description')
  })

  test('The component is well formed so prop moreInfo should exist', () => {
    expect(wrapper.props()).toHaveProperty('moreInfo')
  })

  test('The component is well formed so prop borderVariant should exist', () => {
    expect(wrapper.props()).toHaveProperty('borderVariant')
  })

  test('The component is well formed so prop showCardHeader should exist', () => {
    expect(wrapper.props()).toHaveProperty('showCardHeader')
  })
})
