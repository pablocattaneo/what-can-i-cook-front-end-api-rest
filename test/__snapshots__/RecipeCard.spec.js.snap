// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snaphots Branch: Default. It verify if non-intentional changes were made. 1`] = `
<b-card-stub
  bodytag="div"
  bordervariant="primary"
  class="w-100 h-100"
  footertag="div"
  headertag="div"
  imgalt="Image"
  imgtop="true"
  nobody="true"
  subtitletag="h6"
  subtitletextvariant="muted"
  tag="article"
  titletag="h4"
>
  <!---->
   
  <div
    class="img-wrapper"
  >
    <b-img-lazy-stub
      alt=""
      blankcolor="transparent"
      offset="360"
      src="/"
      title=""
    >
      <a
        variant="primary"
      >
        
      </a>
    </b-img-lazy-stub>
  </div>
   
  <b-card-body-stub
    bodytag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    titletag="h4"
  >
    <b-card-title-stub
      class="text-truncate"
      titletag="h4"
    >
      
    </b-card-title-stub>
     
    <!---->
  </b-card-body-stub>
   
  <!---->
</b-card-stub>
`;

exports[`Snaphots Branch: showCardHeader. It verify if non-intentional changes were made. 1`] = `
<b-card-stub
  bodytag="div"
  bordervariant="primary"
  class="w-100 h-100"
  footertag="div"
  headertag="div"
  imgalt="Image"
  imgtop="true"
  nobody="true"
  subtitletag="h6"
  subtitletextvariant="muted"
  tag="article"
  titletag="h4"
>
  <b-card-header-stub
    class="b-card-header"
    headertag="header"
  >
    <ul
      class="nav nav-pills card-header-pills"
    >
      <li
        class="nav-item"
      >
        <b-button-stub
          tag="button"
          target="_self"
          type="button"
          variant="link"
        >
          
        </b-button-stub>
      </li>
       
      <li
        class="nav-item"
      >
        <b-button-stub
          tag="button"
          target="_self"
          type="button"
          variant="link"
        >
          
        </b-button-stub>
      </li>
    </ul>
  </b-card-header-stub>
   
  <div
    class="img-wrapper"
  >
    <b-img-lazy-stub
      alt=""
      blankcolor="transparent"
      offset="360"
      src="/"
      title=""
    >
      <a
        variant="primary"
      >
        
      </a>
    </b-img-lazy-stub>
  </div>
   
  <b-card-body-stub
    bodytag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    titletag="h4"
  >
    <b-card-title-stub
      class="text-truncate"
      titletag="h4"
    >
      
    </b-card-title-stub>
     
    <!---->
  </b-card-body-stub>
   
  <!---->
</b-card-stub>
`;
